packages/icons/src/components/Icon/index.tsx

```diff
@@ -468,23 +136,53 @@ export const Icon = forwardRef<SVGSVGElement, IconProps>(
       className,
       'data-testid': dataTestId,
       stroke,
+      variant = 'outlined',
       cursor,
       strokeWidth,
       disabled,
     },
     ref,
   ) => {
-    const render = ICONS[name] || ICONS.alert
-
-    const defaultViewBox = useMemo(
-      () =>
-        customViewBoxes.find(vB => vB.icons.includes(name))?.viewBox ??
-        '0 0 24 24',
-      [name],
+    let SystemIcon = StyledIcon(
+      ICONS[
+        variant === 'outlined' && name in ICONS.default ? 'outline' : 'default'
+      ][name] || ICONS.default.alert,
     )
 
+    if (size === 'small' || size === 16) {
+      SystemIcon = StyledIcon(
+        SMALL_ICONS[
+          variant === 'outlined' && name in SMALL_ICONS.default
+            ? 'outline'
+            : 'default'
+        ][name] || SMALL_ICONS.default.alert,
+      )
+    }
```

Comment:
Best is try to avoid using `let` and use only `const` as it encourage immutability, optimization and prevent from doing errors.

Maybe using a `useMemo` and adding if/else condition into it might work